File: bailoteca-backend/src/main/java/com/bailoteca/dto/AsistenciaEventoRequest.java
package com.bailoteca.dto;

import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Clase que representa una solicitud de asistencia a un evento.
 * Contiene información sobre si el usuario asistirá y si ha pagado el evento.
 * 
 * @author Tony Aragón
 * @version 1.0
 * @since 1.0
 */
@Data
@NoArgsConstructor
public class AsistenciaEventoRequest {

    /**
     * Indica si el usuario asistirá al evento.
     */
    private boolean asistira;

    /**
     * Indica si el usuario ha pagado el evento.
     */
    private boolean pagado;
}

File: bailoteca-frontend/app/src/main/java/com/example/bailotecaapp/model/dto/AsistenciaEventoRequest.kt
package com.example.bailotecaapp.model.dto

import kotlinx.serialization.Serializable

/**
 * Representa la petición para registrar o actualizar una asistencia a un evento.
 * Contiene únicamente los datos de estado (asistencia y pago), ya que
 * el usuario y el evento se determinan en el servidor.
 */
@Serializable
data class AsistenciaEventoRequest(
    val asistira: Boolean,
    val pagado: Boolean
)